lang/src/common.rs

```diff
@@ -17,8 +16,11 @@ pub fn close<'info>(
     let mut data = info.try_borrow_mut_data()?;
     let dst: &mut [u8] = &mut data;
     let mut cursor = std::io::Cursor::new(dst);
-    cursor
+    if cursor
```

Comment:
This isn't very rusty. We should favor the functional map_err over the if statement + return.

Reply:
yup thanks. This is a remnant of when I had implemented `error!` to create an `Err` so I couldnt use `map_err` which I then forgot about removing again :p